src/Core/Checkout/Cart/Delivery/Struct/DeliveryDate.php

```diff
@@ -48,7 +49,7 @@ public static function createFromDeliveryTime(DeliveryTime $deliveryTime): self
                 self::create('P' . $deliveryTime->getMin() . 'Y'),
                 self::create('P' . $deliveryTime->getMax() . 'Y')
             ),
-            default => throw new \RuntimeException(\sprintf('Not supported unit %s', $deliveryTime->getUnit())),
+            default => throw CartException::deliveryDateNotSupportedUnit($deliveryTime->getUnit()),
```

Comment:
I would wrap this in a feature flag for BC safety.

```suggestion
            default => Feature::isActive('v6.8.0.0') ? throw CartException::deliveryDateNotSupportedUnit($deliveryTime->getUnit()) : throw new \RuntimeException(\sprintf('Not supported unit %s', $deliveryTime->getUnit())),
```

OR, even better, do the feature flag check in `CartException::deliveryDateNotSupportedUnit()`

Reply:
It is a RuntimeException and anywhere else in our codebase we don't wrap it, also in another PR I did it once and was required to not wrap it

Reply:
not necessary in case of `\RuntimeException`s as the domain exceptions have this as parent deep down :wink:

Reply:
Ah yes true. Apologies.